### ISSUE TITLE
Fedora 26 - NGINX Certbot plugin "not installed"

### ISSUE THREAD

**Issue by joshphp** [certbot/certbot#5132]
## My operating system is (include version): 
Fedora 26
## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
dnf install certbot-nginx

## I ran this command and it produced this output:
sudo certbot --nginx
The requested nginx plugin does not appear to be installed

OR

sudo certbot --nginx certonly
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Could not choose appropriate plugin: The requested nginx plugin does not appear to be installed
The requested nginx plugin does not appear to be installed

## Certbot's behavior differed from what I expected because: 
no cert was issued

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.

2017-09-26 02:51:40,660:DEBUG:certbot.main:certbot version: 0.18.1
2017-09-26 02:51:40,661:DEBUG:certbot.main:Arguments: ['--nginx']
2017-09-26 02:51:40,661:DEBUG:certbot.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#manual,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
2017-09-26 02:51:40,674:DEBUG:certbot.log:Root logging level set at 20
2017-09-26 02:51:40,675:INFO:certbot.log:Saving debug log to /var/log/letsencrypt/letsencrypt.log
2017-09-26 02:51:40,676:DEBUG:certbot.plugins.selection:Requested authenticator nginx and installer nginx
2017-09-26 02:51:40,676:DEBUG:certbot.plugins.selection:No candidate plugin
2017-09-26 02:51:40,676:DEBUG:certbot.plugins.selection:No candidate plugin
2017-09-26 02:51:40,676:DEBUG:certbot.plugins.selection:Selected authenticator None and installer None
2017-09-26 02:51:40,676:INFO:certbot.main:Could not choose appropriate plugin: The requested nginx plugin does not appear to be installed
2017-09-26 02:51:40,677:DEBUG:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/bin/certbot", line 11, in <module>
    load_entry_point('certbot==0.18.1', 'console_scripts', 'certbot')()
  File "/usr/lib/python3.6/site-packages/certbot/main.py", line 755, in main
    return config.func(config, plugins)
  File "/usr/lib/python3.6/site-packages/certbot/main.py", line 673, in certonly
    installer, auth = plug_sel.choose_configurator_plugins(config, plugins, "certonly")
  File "/usr/lib/python3.6/site-packages/certbot/plugins/selection.py", line 201, in choose_configurator_plugins
    diagnose_configurator_problem("authenticator", req_auth, plugins)
  File "/usr/lib/python3.6/site-packages/certbot/plugins/selection.py", line 297, in diagnose_configurator_problem
    raise errors.PluginSelectionError(msg)
certbot.errors.PluginSelectionError: The requested nginx plugin does not appear to be installed

**Comment by bmw** [comment]
Thanks for reporting this!

Looks like what's going on is when you run `dnf install certbot-nginx`, it installs both the Python 2 and Python 3 version of Certbot but only the Python 2 version of the Nginx plugin. Fedora has configured the executable "certbot" to refer to the Python 3 version where the Nginx plugin wasn't installed.

I've reported this [here](https://bugzilla.redhat.com/show_bug.cgi?id=1497315).

In the meantime, you can either:

1. Use the executable "certbot-2".
2. Run `dnf install python3-certbot-nginx`.

I'm closing this issue in favor of tracking this bug in the Fedora package on Bugzilla, but let me know if there's anything else I can do to help.

**Comment by joshphp** [comment]
@bmw thank you, certbo2 worked perfectly!

**Comment by andreasg123** [comment]
I ran into a similar issue where `certbot renew -q --no-self-upgrade` on Fedora 26 stopped working after `dnf update` with the same error message:
The requested apache plugin does not appear to be installed

I tried this:

```
dnf install certbot-apache
Package python2-certbot-apache-0.18.1-1.fc26.noarch is already installed, skipping.
```

Based on the comments here, I ran `dnf install python3-certbot-apache`.  That fixed my problem.

It looks like the recent `dnf update` switched to Python 3 without giving any indications.
